src/plugins/saved_objects/README.md

```diff
@@ -0,0 +1,119 @@
+# Saved Object
+
+The saved object plugin provides all the core services and functionalities of the saved objects, and is utilized by many other plugins such as visualizaton plugin, dashboard plugin and wizard plugin etc.
+
+## Save Relationships to Index Pattern
+
+The relationships to index patterns are saved using the kibanaSavedObjectMeta attribute and the references array structure. Functions from the data plugin are utilized by the saved object plugin to manage this index pattern relationships. 
+
+A standard saved object and its index pattern relationship:
+
+```.ts
+
+"kibanaSavedObjectMeta" : {
+    "searchSourceJSON" : """{"filter":[],"query":{"query":"","language":"kuery"},"indexRefName":"kibanaSavedObjectMeta.searchSourceJSON.index"}"""
+      }
+    },
+    "type" : "visualization",
+    "references" : [
+      {
+          "name" : "kibanaSavedObjectMeta.searchSourceJSON.index",
+          "type" : "index-pattern",
+          "id" : "90943e30-9a47-11e8-b64d-95841ca0b247"
+      }
+    ],
+
+```
+
+### Saving a saved object
```

Comment:
Style nit - for headers, be consistent with casing, at least in the same doc, so "All Title Case" headers or "All sentence case" headers. My personal preference is for sentence case.

Reply:
Change to all sentence case headers~